Docs fixing and cleanup (#1217)
netket/netket: docs/docs/varstate.md

## Constructing a Variational State

To construct the two variational states above you need to provide at least a Monte Carlo sampler (you can find the list of available ones [here](netket_sampler_api)) and a model.

> The hilbert space of the variational state will be inferred from the sampler, as they all reference the hilbert space they are sampling.

Reviewer: ```suggestion
The Hilbert space of the variational state will be inferred from the sampler, which contains a reference to the underlying Hilbert space.
```
